src/ship.js

```diff
@@ -0,0 +1,45 @@
+function Ship (itinerary) {
+  this.itinerary = itinerary;
+  this.currentPort = itinerary.ports[0];
+  this.previousPort = null
+  this.currentPort.addShip(this);
+}
+
+Ship.prototype = {
+  setSail() {
+    const itinerary = this.itinerary;
+    const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+  
+    if (currentPortIndex === (itinerary.ports.length - 1)) {
+      throw new Error('End of itinerary reached');
+    }
+  
+    
+    this.previousPort = this.currentPort;
+    this.currentPort = null;
+    this.previousPort.removeShip(this);
+    },
```

Comment:
The closing brace should be at the same level as the opening brace of the function:
```
  },
```